osu.Game/Online/API/Requests/Responses/APIScoreInfo.cs

```diff
@@ -81,57 +142,8 @@ public ScoreInfo CreateScoreInfo(RulesetStore rulesets)
             return scoreInfo;
         }
 
-        [JsonProperty(@"score")]
-        public int TotalScore { get; set; }
-
-        [JsonProperty(@"max_combo")]
-        public int MaxCombo { get; set; }
-
-        [JsonProperty(@"user")]
-        public User User { get; set; }
-
-        [JsonProperty(@"id")]
-        public long OnlineScoreID { get; set; }
-
-        [JsonProperty(@"replay")]
-        public bool Replay { get; set; }
-
-        [JsonProperty(@"created_at")]
-        public DateTimeOffset Date { get; set; }
+        public IRulesetInfo Ruleset => new RulesetInfo { ID = OnlineRulesetID };
```

Comment:
These local `RulesetInfo` instantiations worry me somewhat because `CreateInstance()` is declared on the interface, and it won't work as expected here due to the missing `InstantiationInfo` spec. I don't know what the solution is, though. Probably `CreateInstance()` should only be callable via the databased variants?

Reply:
This is also an issue with `IMod` interface. We may eventually want to move `CreateInstance` to the databased types only. Not sure if that needs to be resolved in this PR.

If this is seen as an issue, we can replace this with a raw `RulesetID`, though I wrote it this way to avoid changing the underlying interfaces too much (ie. making `IScoreInfo` match `IBeatmapInfo` etc.)